### ISSUE TITLE
 Issue with Reproducing Results in evaluate_depth.py

### ISSUE THREAD

**Issue by keaisg** [BeileiCui/EndoDAC#7]
Thank you for your valuable work. I am currently trying to reproduce the results from your paper using the provided codebase. However, I am encountering significantly different results when running evaluate_depth.py.

- OS: Ubuntu 22.04.2
- Python Version : python 3.9.19
- Package Version :torch ==2.0.0 torchvision==0.15.0

**Command Used:**`CUDA_VISIBLE_DEVICES=0 python evaluate_depth.py --data_path /media/zxk/data/endovis_data --load_weights_folder './EndoDAC_fullmodel' --eval_mono`
**Output:** `/home/zxk/cbs/EndoDACv2/models/backbones/layers/swiglu_ffn.py:43: UserWarning: xFormers is available (SwiGLU)
  warnings.warn("xFormers is available (SwiGLU)")
/home/zxk/cbs/EndoDACv2/models/backbones/layers/attention.py:27: UserWarning: xFormers is available (Attention)
  warnings.warn("xFormers is available (Attention)")
/home/zxk/cbs/EndoDACv2/models/backbones/layers/block.py:33: UserWarning: xFormers is available (Block)
  warnings.warn("xFormers is available (Block)")
-> Loading weights from ./EndoDAC_fullmodel
load pretrained weight from /home/zxk/cbs/EndoDACv2/pretrained_model/depth_anything_vitb14.pth

-> Computing predictions with size 320x256
345it [00:11, 34.66it/s]
/home/zxk/anaconda3/envs/endodac/lib/python3.9/site-packages/numpy/core/fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
/home/zxk/anaconda3/envs/endodac/lib/python3.9/site-packages/numpy/core/_methods.py:129: RuntimeWarning: invalid value encountered in divide
  ret = ret.dtype.type(ret / rcount)
/home/zxk/cbs/EndoDACv2/utils/utils.py:115: RuntimeWarning: Mean of empty slice.
  a1 = (thresh < 1.25     ).mean()
/home/zxk/anaconda3/envs/endodac/lib/python3.9/site-packages/numpy/core/_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
/home/zxk/cbs/EndoDACv2/utils/utils.py:116: RuntimeWarning: Mean of empty slice.
  a2 = (thresh < 1.25 ** 2).mean()
/home/zxk/cbs/EndoDACv2/utils/utils.py:117: RuntimeWarning: Mean of empty slice.
  a3 = (thresh < 1.25 ** 3).mean()
/home/zxk/cbs/EndoDACv2/utils/utils.py:120: RuntimeWarning: Mean of empty slice.
  rmse = np.sqrt(rmse.mean())
/home/zxk/cbs/EndoDACv2/utils/utils.py:123: RuntimeWarning: Mean of empty slice.
  rmse_log = np.sqrt(rmse_log.mean())
551it [00:18, 30.41it/s]
 Scaling ratios | med: 54.563 | std: 0.578

           abs_rel      |      sq_rel      |        rmse      |    rmse_log      |          a1      |          a2      |          a3      | 
mean:&       4.049      &      50.244      &       8.411      &       1.004      &       0.252      &       0.471      &       0.665      \\
cls: & [ 3.945,  4.153] & [ 47.833,  52.655] & [ 8.087,  8.735] & [ 0.985,  1.023] & [ 0.241,  0.264] & [ 0.460,  0.483] & [ 0.656,  0.674] \\
average inference time: 6.4 ms

-> Done!
`
**Problem Description:**
When running evaluate_depth.py, I expected to obtain results consistent with those reported in your paper. However, the metrics I am obtaining are quite different, particularly in terms of RMSE and related metrics.

**Dataset Structure:**
Here is the structure of my dataset that I am using for evaluation:
`-endovis_data/
    -test/
       -dataset8/
       -dataset9/
    -train/
       -dataset1/
            -keyframe1/
                -data/
                   -frame_data/
                        -frame_data000000.json
                   -left/
                        -0000000001.png
                   -scene_points/
                        -scene_points000000.tiff
            -keyframe2/
                -data/
                   -frame_data/
                   -left/
                   -scene_points/
       -dataset2/
       -dataset3/
       -dataset4/
       -dataset5/
       -dataset6/
       -dataset7/
       -dataset8/
       -dataset9/
`

**Modifications Made:**
I made the following changes in evaluate_depth.py and scared_dataset.py:

In evaluate_depth.py:
`# cl = st.t.interval(alpha=0.95, df=len(errors) - 1, loc=mean_errors[i], scale=st.sem(errors[:, i]))        
cl = st.t.interval(confidence=0.95, df=len(errors)-1, loc=mean_errors[i], scale=st.sem(errors[:,i]))
`
In scared_dataset.py:
`# f_str = "frame_data{:06d}{}".format(frame_index-1, self.img_ext)        
f_str = "{:010d}{}".format(frame_index - 1, self.img_ext)
`

**Request for Assistance:**
Could you please provide guidance on how to resolve this discrepancy? Any insights or suggestions would be greatly appreciated.

Thank you once again for your assistance.

**Comment by keaisg** [comment]
I have solved this issue.

**Comment by Girin325** [comment]
> I have solved this issue.

Could you help me that how can you solve that?
